### ISSUE TITLE
Error: Doctrine\DBAL\Query\QueryException: More than 1000 expressions in a list are not allowed on Oracle.

### ISSUE THREAD

**Issue by sebidotorg** [nextcloud/server#27932]
### Steps to reproduce
Enter a folder containing around 1500 photos (JPG, RAW, and XMP). The error message quoted below immediately appears in the log.

### Expected behaviour
No error. There is no Oracle on this system.

### Actual behaviour
Level: Error
App: core
Message: Doctrine\DBAL\Query\QueryException: More than 1000 expressions in a list are not allowed on Oracle.

## Server configuration detail

**Operating system:** Linux 5.10.46 #1 SMP Fri Jun 25 01:51:18 CEST 2021 x86_64

**Webserver:** Apache (fpm-fcgi)

**Database:** mysql 10.3.29

**PHP version:** 

7.4.15
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, sodium, cgi-fcgi, mysqlnd, PDO, xml, bcmath, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, intl, json, exif, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, soap, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlrpc, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 21.0.3 - 21.0.3.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated since NC 10(?), always running the latest version from stable.

**Where did you install Nextcloud from:** Download from nextcloud.com.

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.7.0
 - activity: 2.14.3
 - admin_audit: 1.11.0
 - bruteforcesettings: 2.2.0
 - calendar: 2.3.0
 - camerarawpreviews: 0.7.11
 - checksum: 1.1.2
 - cloud_federation_api: 1.4.0
 - comments: 1.11.0
 - contacts: 4.0.0
 - contactsinteraction: 1.2.0
 - dashboard: 7.1.0
 - dav: 1.17.1
 - encryption: 2.9.0
 - extract: 1.3.2
 - federatedfilesharing: 1.11.0
 - federation: 1.11.0
 - files: 1.16.0
 - files_antivirus: 3.2.1
 - files_external: 1.12.0
 - files_markdown: 2.3.4
 - files_mindmap: 0.0.24
 - files_pdfviewer: 2.1.0
 - files_rightclick: 1.0.0
 - files_sharing: 1.13.1
 - files_texteditor: 2.14.0
 - files_trashbin: 1.11.0
 - files_versions: 1.14.0
 - files_videoplayer: 1.10.0
 - firstrunwizard: 2.10.0
 - forms: 2.2.4
 - imageconverter: 1.3.1
 - issuetemplate: 0.7.0
 - logreader: 2.6.0
 - lookup_server_connector: 1.9.0
 - metadata: 0.14.0
 - music: 1.2.1
 - nextcloud_announcements: 1.10.0
 - notifications: 2.9.0
 - oauth2: 1.9.0
 - password_policy: 1.11.0
 - photos: 1.3.0
 - privacy: 1.5.0
 - provisioning_api: 1.11.0
 - quota_warning: 1.10.0
 - rainloop: 7.1.2
 - ransomware_protection: 1.11.0
 - recommendations: 1.0.0
 - richdocuments: 4.2.0
 - richdocumentscode: 6.4.1003
 - serverinfo: 1.11.0
 - settings: 1.3.0
 - sharebymail: 1.11.0
 - spreed: 11.2.2
 - support: 1.4.0
 - survey_client: 1.9.0
 - systemtags: 1.11.0
 - tasks: 0.13.6
 - text: 3.2.0
 - theming: 1.12.0
 - twofactor_backupcodes: 1.10.0
 - twofactor_nextcloud_notification: 3.1.2
 - twofactor_totp: 6.1.0
 - updatenotification: 1.11.0
 - user_status: 1.1.1
 - user_usage_report: 1.5.0
 - viewer: 1.5.0
 - weather_status: 1.1.0
 - workflow_ocr: 1.21.0
 - workflowengine: 2.3.0
Disabled:
 - nextbackup
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.sebi.org"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "htaccess.RewriteBase": "\/",
    "overwrite.cli.url": "https:\/\/cloud.sebi.org\/",
    "version": "21.0.3.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "3306",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "default_phone_region": "de",
    "installed": true,
    "mail_smtpmode": "sendmail",
    "mail_smtpauthtype": "PLAIN",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "filelocking.enabled": true,
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0
    },
    "maintenance": false,
    "loglevel": 1,
    "updater.release.channel": "stable",
    "theme": "",
    "data-fingerprint": "8cadcd3acd11b446973847d9e426736b",
    "twofactor_enforced": "true",
    "twofactor_enforced_groups": [
        "admin"
    ],
    "twofactor_enforced_excluded_groups": [],
    "mysql.utf8mb4": true,
    "logtimezone": "Europe\/Berlin",
    "app_install_overwrite": [
        "files_ebookreader",
        "files_reader",
        "ransomware_detection",
        "ocr",
        "emlviewer",
        "issuetemplate",
        "nextbackup"
    ],
    "mail_sendmailmode": "pipe",
    "enable_previews": true,
    "enabledPreviewProviders": [
        "OC\\Preview\\TXT",
        "OC\\Preview\\MarkDown",
        "OC\\Preview\\OpenDocument",
        "OC\\Preview\\PDF",
        "OC\\Preview\\Image",
        "OC\\Preview\\Photoshop",
        "OC\\Preview\\HEIC",
        "OC\\Preview\\MP3"
    ],
    "preview_max_x": "1920",
    "preview_max_y": "1920"
}
```
</details>

**Are you using external storage, if yes which one:** One SFTP directory, not involved in this error.

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** none

## Client configuration

**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15

**Operating system:** 

## Logs

<details><summary>Web server error log</summary>

```

```
</details>

<details><summary>Nextcloud log</summary>

```
{"reqId":"MSsWCzODicQnwaFPdFBX","level":3,"time":"2021-07-13T03:27:27+02:00","remoteAddr":"2a02:8071:1a9:b500:f494:4f8b:4978:d9b4","user":"sebi","app":"core","method":"PROPFIND","url":"/remote.php/dav/files/sebi/Harald/Pictures/Familie%20und%20Urlaube/Familienbilder%20Repros%20ab%201977","message":{"Exception":"Doctrine\\DBAL\\Query\\QueryException","Message":"More than 1000 expressions in a list are not allowed on Oracle.","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/Comments/Manager.php","line":656,"function":"execute","class":"OC\\DB\\QueryBuilder\\QueryBuilder","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/CommentPropertiesPlugin.php","line":89,"function":"getNumberOfUnreadCommentsForObjects","class":"OC\\Comments\\Manager","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/CommentPropertiesPlugin.php","line":117,"function":"cacheDirectory","class":"OCA\\DAV\\Connector\\Sabre\\CommentPropertiesPlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"handleGetProperties","class":"OCA\\DAV\\Connector\\Sabre\\CommentPropertiesPlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1052,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":984,"function":"getPropertiesByNode","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1661,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1646,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Server.php","line":332,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/remote.php","line":167,"args":["/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php","Line":268,"CustomMessage":"More than 1000 expressions in a list are not allowed on Oracle."},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15","version":"21.0.3.1","id":"60ecee9aa5c49"}
```
</details>

<details><summary>Browser log</summary>

</details>

**Comment by andyxheli** [comment]
duplicate of https://github.com/nextcloud/server/issues/26493 still waiting on backport to  NC 21

**Comment by szaimen** [comment]
Closing based on andxheli's comment
